scene/resources/sprite_frames.cpp

```diff
@@ -175,8 +201,16 @@ void SpriteFrames::_set_animations(const Array &p_animations) {
 		anim.loop = d["loop"];
 		Array frames = d["frames"];
 		for (int j = 0; j < frames.size(); j++) {
-			Ref<Resource> res = frames[j];
-			anim.frames.push_back(res);
+			Dictionary f = frames[j];
```

Comment:
Should I add texture support here instead of a dictionary (for partial 3.x compatibility)? Or is it a issue for the converter (and a separate PR)?

Reply:
Ah yes, Dictionary type is not good for performance and it would be better to change it.

Reply:
Such as
```cpp
	struct Anim {
		double speed = 5.0;
		bool loop = true;
		Vector<Ref<Texture2D>> frames;
		Vector<Ref<Texture2D>> durations;
	};
```

Reply:
> Ah yes, Dictionary type is not good for performance and it would be better to change it.

This property is for serialization only.

> ```c++
> 	struct Anim {
> 		double speed = 5.0;
> 		bool loop = true;
> 		Vector<Ref<Texture2D>> frames;
> 		Vector<float> durations;
> 	};
> ```

The internal structure does not have to match the serialization format (individual durations require additional code when renaming or deleting animations).

Reply:
Oh sorry I misunderstood, the Dictionary type has been used in here before, so I think it's not a problem.
I think the current struct Frame is fine, but the compatibility breakdown is probably due to a change somewhere in this PR, either in the name of the property or in the way it is stored.

Reply:
Then, if there are unresolvable reasons that cannot be avoided, you can modify the 3 to 4 converter.

Reply:
Before:

```
[sub_resource type="SpriteFrames" id="ID"]
animations = [{
"frames": [SubResource("ID"), SubResource("ID")],
"loop": true,
"name": &"default",
"speed": 5.0
}]
```

After:


```
[sub_resource type="SpriteFrames" id="ID"]
animations = [{
"frames": [{
"duration": 2.5,
"texture": ExtResource("ID")
}, {
"duration": 1.0,
"texture": ExtResource("ID")
},
"loop": true,
"name": &"default",
"speed": 5.0
}]
```

This PR breaks backwards compatibility anyway (because not only the serialization format changes, but also the public API). I will try to add changes to the converter after this PR is accepted and merged.

Reply:
Done. Now compatible with the old serialization format.

I will add the following replacements for the converter in a separate PR:

```
get_frame -> get_frame_texture
add_frame(a, b, c) -> add_frame(a, b, 1.0, c)
```